Tests/Provider/FileProviderTest.php

```diff
@@ -224,4 +224,85 @@ public function mediaProvider()
             array(null, $media3),
         );
     }
+
+    public function testBinaryContentWithRealPath()
+    {
+        $media = $this->getMock('Sonata\MediaBundle\Model\MediaInterface');
+
+        $media->expects($this->any())
+            ->method('getProviderReference')
+            ->willReturn('provider');
+
+        $media->expects($this->any())
+            ->method('getId')
+            ->willReturn(10000);
+
+        $media->expects($this->any())
+            ->method('getContext')
+            ->willReturn('context');
+
+        $binaryContent = $this->getMockBuilder('Symfony\Component\HttpFoundation\File\File')
+            ->setMethods(array('getRealPath', 'getPathname'))
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $binaryContent->expects($this->atLeastOnce())
+            ->method('getRealPath')
+            ->willReturn(__DIR__.'/../fixtures/file.txt');
+
+        $binaryContent->expects($this->never())
+            ->method('getPathname');
+
+        $media->expects($this->any())
+            ->method('getBinaryContent')
+            ->willReturn($binaryContent);
+
+        $provider = $this->getProvider();
+
+        $setFileContents = new \ReflectionMethod('Sonata\MediaBundle\Provider\FileProvider', 'setFileContents');
+        $setFileContents->setAccessible(true);
+
+        $setFileContents->invoke($provider, $media);
```

Comment:
If there were pre-existing tests to check the behavior of setFileContents (via the public postPersist/postUpdate), I would have gladly added to them. However, there aren't. These tests seemed the most robust way to check the behavior introduced by my PR given the current state of the repo. Although creating unit tests for postPersist and postUpdate would be the perfect solution, it is clearly outside the scope of this PR.